attest/index.js

```diff
@@ -0,0 +1,80 @@
+import core from '@actions/core';
+import exec from '@actions/exec';
+import artifact from '@actions/artifact';
+import path from 'path';
+
+function getActionConfig() {
+    return {
+        cimon: {
+            logLevel: core.getInput('logLevel'),
+            clientId: core.getInput('clientId'),
+            secret: core.getInput('secret'),
+            url: core.getInput('url'),
+            releasePath: core.getInput('releasePath'),
+        },
+        attest: {
+            subjects: core.getInput('subjects'),
+            imageRef: core.getInput('imageRef'),
+            signKey: core.getInput('signKey'),
+            provenanceOutput: core.getInput('provenanceOutput'),
+            signedProvenanceOutput: core.getInput('signedProvenanceOutput'),
+            githubContext: core.getInput('githubContext'),
+        },
+        report: {
+            reportJobSummary: core.getInput('reportJobSummary'),
+            reportArtifact: core.getInput('reportArtifact'),
+        }
+    };
+}
+  
+async function run() {
+    try {
+        const config = getActionConfig();
+
+        const env = {
+            ...process.env,
+            CIMON_SUBJECTS: config.attest.subjects,
+            CIMON_ATTEST_IMAGE_REF: config.attest.imageRef,
+            CIMON_SIGN_KEY: config.attest.signKey,
+            CIMON_PROVENANCE_OUTPUT: config.attest.provenanceOutput,
+            CIMON_SIGNED_PROVENANCE_OUTPUT: config.attest.signedProvenanceOutput,
+            GITHUB_CONTEXT: config.attest.githubContext,
+            CIMON_LOG_LEVEL: config.cimon.logLevel,
+            CIMON_CLIENT_ID: config.cimon.clientId,
+            CIMON_SECRET: config.cimon.secret,
+            CIMON_URL: config.cimon.url,
+            CIMON_REPORT_JOB_SUMMARY: config.report.reportJobSummary,
+            CIMON_REPORT_ARTIFACT: config.report.reportArtifact,
+        };
+
+        const options = {
+            env,
+            silent: true,
+        };
+        
+        const scriptPath = path.join(__dirname, 'attest.sh');
+        const exitCode = await exec.exec('bash', [scriptPath, config.cimon.releasePath], options);
+        if (exitCode !== 0) {
+            throw new Error('Failed creating provenance');
+        }
+
+        if (config.report.reportArtifact.toLowerCase() === 'true') {
```

Comment:
See comment above. Once it's converted, could be checked with `if (config.report.reportArtifact) {...`